.github/workflows/ci.yml

```diff
@@ -0,0 +1,95 @@
+name: RSpec CI
+on:
+  push:
+    branches:
+      - 'main'
+      - '*-maintenance'
+      - '*-dev'
+  pull_request:
+    branches:
+      - '*'
+jobs:
+  test:
+    name: 'Ruby: ${{ matrix.ruby }}, Rails: ${{ matrix.env.RAILS_VERSION }}'
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+       include:
+         # Rails 6.0 builds >= 2.5.0
+         - ruby: 3.0.0-preview1
+           allow_failure: true
+           env:
+             RAILS_VERSION: '~> 6.0.0'
+         - ruby: 2.7
+           env:
+             RAILS_VERSION: '~> 6.0.0'
+         - ruby: 2.6
+           env:
+             RAILS_VERSION: '~> 6.0.0'
+         - ruby: 2.5
+           env:
+             RAILS_VERSION: '~> 6.0.0'
+         # Rails 5.2 builds >= 2.2.2
+         - ruby: 2.6.6
+           env:
+             RAILS_VERSION: '~> 5.2.0'
+         - ruby: 2.5.8
+           env:
+             RAILS_VERSION: '~> 5.2.0'
+         - ruby: 2.4.10
+           env:
+             RAILS_VERSION: '~> 5.2.0'
+         - ruby: 2.3.8
+           env:
+             RAILS_VERSION: '~> 5.2.0'
+         - ruby: 2.2.10
+           allow_failure: true
+           env:
+             RAILS_VERSION: '~> 5.2.0'
+         - ruby: 2.2.10
+           env:
+             RAILS_VERSION: '5-2-stable'
+         # Rails 5.1 Builds >= 2.2.2
+         - ruby: 2.6.6
+           env:
+             RAILS_VERSION: '~> 5.1.0'
+         - ruby: 2.5.8
+           env:
+             RAILS_VERSION: '~> 5.1.0'
+         - ruby: 2.4.10
+           env:
+             RAILS_VERSION: '~> 5.1.0'
+         - ruby: 2.3.8
+           env:
+             RAILS_VERSION: '~> 5.1.0'
+         - ruby: 2.2.10
+           env:
+             RAILS_VERSION: '~> 5.1.0'
+         # Rails 5.0 Builds >= 2.2.2
+         - ruby: 2.6.6
+           env:
+             RAILS_VERSION: '~> 5.0.0'
+         - ruby: 2.5.8
+           env:
+             RAILS_VERSION: '~> 5.0.0'
+         - ruby: 2.4.10
+           env:
+             RAILS_VERSION: '~> 5.0.0'
+         - ruby: 2.3.8
+           env:
+             RAILS_VERSION: '~> 5.0.0'
+         - ruby: 2.2.10
+           env:
+             RAILS_VERSION: '~> 5.0.0'
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: false
+      - run: script/update_rubygems_and_install_bundler
+      - run: script/clone_all_rspec_repos
+      - run: bundle install --binstubs && script/run_build
+        env: ${{ matrix.env }}
```

Comment:
I had it only here, not sure it is needed before (CI pass). I saw Jon that you add it to all steps.

Reply:
You need it for each step where it's expected, I'm looking into a similar issue to you with versions.